### PR TITLE
Fix ssh tunnel caching.

### PR DESCRIPTION
**Pull request by rohinb2**
modifies some of the changes landed by @jlewitt1 in https://app.graphite.dev/github/pr/run-house/runhouse/166/SageMaker-Cluster-updates?utm_source=stack-comment

### CODE REVIEW

**Comment by rohinb2** [comment]
Current dependencies on/for this PR:
* main
  * **PR #190** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/190?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #191** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/191?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/191?utm_source=stack-comment).

File: runhouse/resources/hardware/cluster.py [hunk 1]
```diff
@@ -519,13 +519,14 @@ def ssh_tunnel(
         # netstat -vanp tcp | grep 32300
         # lsof -i :32300
         # kill -9 <pid>
-        open_tunnels = open_cluster_tunnels.get((self.address, self.ssh_port), [])
-        for (tunnel, port) in open_tunnels:
+        open_tunnel = open_cluster_tunnels.get((self.address, self.ssh_port))
+        if open_tunnel is not None:
+            tunnel, port = open_tunnel
```
Comment (dongreenberg) [inline runhouse/resources/hardware/cluster.py:524]: Am I dense that I don't understand the difference between these

**Comment by rohinb2** [comment]
we don't store a list in the `open_cluster_tunnels`, we store a tuple

**Review by dongreenberg — APPROVED** [review]
Oh!

**Pull request merged**